oslo_vmware/rw_handles.py

```diff
@@ -310,30 +310,44 @@ def __init__(self, session, lease, url, file_handle,
 
     def _log_progress(self, progress):
         """Log data transfer progress."""
-        if (progress == 100 or (progress - self._last_logged_progress >=
-                                MIN_PROGRESS_DIFF_TO_LOG)):
-            LOG.debug("Data transfer progress is %d%%.", progress)
-            self._last_logged_progress = progress
+        if 'bytes_read' in progress:
+            metric = progress["bytes_read"]
+        else:
+            metric = progress["bytes_written"]
+
+        progress["metric"] = metric
+        time_diff = time.time() - self._last_progress_update
+        progress["time_diff"] = datetime.timedelta(seconds=time_diff)
+
+        LOG.info("Data Transfer progress: %(metric)s"
+                " of %(total_size)s in %(time_diff)s", progress)
 
     def _check_progress_stalled(self, progress, now):
         """Check if we still make progress
 
         Compares the given progress to the last known progress to see if we
         still go forward. Emits a warning otherwise.
         """
-        if progress < self._last_progress:
+        if 'bytes_read' in progress:
+            metric = progress["bytes_read"]
+        else:
+            metric = progress["bytes_written"]
+
+        if metric < self._last_logged_progress:
             LOG.warning('We made negative progress: before %(before)s%%, '
                         'now %(now)s%%',
                         {'before': self._last_progress,
-                         'now': progress})
+                         'now': progress['percent']})
             self._last_progress_change = now
-        elif progress == self._last_progress:
+        elif metric == self._last_logged_progress:
```

Comment:
This will always hit on first run and emit a warning. We don't want that. I proposed this on slack:
```diff
$ git diff
diff --git a/oslo_vmware/rw_handles.py b/oslo_vmware/rw_handles.py
index 28ef83c..ae6113e 100644
--- a/oslo_vmware/rw_handles.py
+++ b/oslo_vmware/rw_handles.py
@@ -298,7 +298,7 @@ class VmdkHandle(FileHandle):
         self._last_logged_progress = 0
         self._last_progress_udpate = 0
         self._last_progress = 0
-        self._last_progress_change = 0
+        self._last_progress_change = time.time()
         self._updater = None
         if update_progress:
@@ -327,7 +327,10 @@ class VmdkHandle(FileHandle):
                         {'before': self._last_progress,
                          'now': progress})
             self._last_progress_change = now
-        elif progress == self._last_progress and self._last_progress_change:
+        elif (progress == self._last_progress
+                and (progress != 0 or now - self._last_progress_change > 300)):
+            # we only want this warning if we moved away from 0 or if we ran at
+            # least 5min
             LOG.warning('No progress made in %(interval)ss. Currently at '
                         '%(progress)s%%',
                         {'interval': now - self._last_progress_change,
```